### PR TITLE
i18n(es): Translate `site-search`

### PR DESCRIPTION
**Pull request by dreyfus92**
#### Description

- What does this PR change? Give us a brief description.
Translated `site-search` to spanish

### CODE REVIEW

**Review by Villanuevand — APPROVED** [review]

File: docs/src/content/docs/es/guides/site-search.mdx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+Por defecto, los sitios de Starlight incluyen una búsqueda de texto completo impulsada por [Pagefind](https://pagefind.app/), que es una herramienta de búsqueda rápida y de bajo ancho de banda para sitios estáticos.
+
+No es necesaria ninguna configuración para habilitar la búsqueda. Construye e implementa tu sitio, luego usa la barra de búsqueda en el encabezado del sitio para encontrar contenido.
+
+## Ocultar contenido en los resultados de búsqueda
+
+### Excluye una página
+
+Para excluir una página desde el índice de búsqueda, agrega [`pagefind: false`](/reference/frontmatter/#pagefind) al frontmatter de la página:
```
Comment (kevinzunigacuellar) [inline docs/src/content/docs/es/guides/site-search.mdx:18]:
```suggestion
Para excluir una página del índice de búsqueda, agrega [`pagefind: false`](/reference/frontmatter/#pagefind) al frontmatter de la página:
```

File: docs/src/content/docs/es/guides/site-search.mdx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+Este texto será descubrible a través de la búsqueda.
+
+<div data-pagefind-ignore>
+
+Este texto estará oculto de la búsqueda.
+
+</div>
+```
+
+## Alternativa de proveedores de búsqueda
```
Comment (kevinzunigacuellar) [inline docs/src/content/docs/es/guides/site-search.mdx:47]:
```suggestion
## Proveedores de búsqueda alternativos
```

**Review by kevinzunigacuellar — COMMENTED** [review]
Thanks @dreyfus92 💜 , I left a few comments for better legibility.

**Comment by dreyfus92** [comment]
Muchas gracias por la revisión @Villanuevand y @kevinzunigacuellar 😁

**Pull request merged**